examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -1032,162 +918,54 @@ def collate_fn(examples):
+    #    # create custom saving & loading hooks so that `accelerator.save_state(...)` serializes in a nice format
+    #    def save_model_hook(models, weights, output_dir):
+    #        if accelerator.is_main_process:
+    #            if args.use_ema:
+    #                ema_unet.save_pretrained(os.path.join(output_dir, "unet_ema"))
 
-            text_encoder_state_dict = {k: v for k, v in lora_state_dict.items() if "text_encoder." in k}
-            LoraLoaderMixin.load_lora_into_text_encoder(
-                text_encoder_state_dict, network_alphas=network_alphas, text_encoder=text_encoder_one_
-            )
+    #            for i, model in enumerate(models):
+    #                model.save_pretrained(os.path.join(output_dir, "unet"))
 
-            text_encoder_2_state_dict = {k: v for k, v in lora_state_dict.items() if "text_encoder_2." in k}
-            LoraLoaderMixin.load_lora_into_text_encoder(
-                text_encoder_2_state_dict, network_alphas=network_alphas, text_encoder=text_encoder_two_
-            )
+    #                # make sure to pop weight so that corresponding model is not saved again
+    #                weights.pop()
 
-        accelerator.register_save_state_pre_hook(save_model_hook)
-        accelerator.register_load_state_pre_hook(load_model_hook)
-    else:
-        # Set unet as trainable.
-        unet.train()
+    #    def load_model_hook(models, input_dir):
+    #        if args.use_ema:
+    #            load_model = EMAModel.from_pretrained(os.path.join(input_dir, "unet_ema"), UNet2DConditionModel)
+    #            ema_unet.load_state_dict(load_model.state_dict())
+    #            ema_unet.to(accelerator.device)
+    #            del load_model
 
-        # For mixed precision training we cast all non-trainable weigths to half-precision
-        # as these weights are only used for inference, keeping weights in full precision is not required.
+    #        for i in range(len(models)):
+    #            # pop models so that they are not loaded again
+    #            model = models.pop()
 
-        weight_dtype = torch.float32
+    #            # load diffusers style into model
+    #            load_model = UNet2DConditionModel.from_pretrained(input_dir, subfolder="unet")
+    #            model.register_to_config(**load_model.config)
 
-        if gaudi_config.use_torch_autocast or args.bf16:
-            weight_dtype = torch.bfloat16
+    #            model.load_state_dict(load_model.state_dict())
+    #            del load_model
 
-        # Move unet, vae and text_encoder to device and cast to weight_dtype
-        # The VAE is in float32 to avoid NaN losses.
-        vae.to(accelerator.device, dtype=weight_dtype)
-        text_encoder_one.to(accelerator.device, dtype=weight_dtype)
-        text_encoder_two.to(accelerator.device, dtype=weight_dtype)
-        del text_encoders, tokenizers
-        gc.collect()
-        # Create EMA for the unet.
-        if args.use_ema:
```

Comment:
should we enable 1151-1189?

Reply:
not clear which portion you meant. can you please specify more details?